epcis_event_hash_generator/hash_generator.py

```diff
@@ -180,8 +180,8 @@ def _gather_elements_not_in_order(children, child_order):
 
     # remove fields that are to be ignored in the hash:
     # remove all elements from XML tree which do shouldn't take part in hash calculation
-    to_be_ignored = ["recordTime", "eventID"]
-    for child in children:
+    to_be_ignored = ["recordTime", "eventID", "type"]
+    for child in children[:]:
```

Comment:
With the exception of this expression, all your changes look good to me, @dakbhavesh. **Thanks A LOT** for making all these adjustments!
I didn't develop sth. in Python for a couple of months now (it may be a new gimmick I am not aware of yet), but shouldn't be the for loop just expressed as 
`for child in children: `instead of `for child in children[:]:`?

Reply:
Hi @RalphTro, I am also not seasoned in python and trying to get my hands-on whenever I find an opportunity :). In this particular change, I had to change `child in children` to  `child in children[:]` because the loop was breaking after the removal of the first tuple in the list and prehash text was containing the unwanted entry e.g. type=ObjectEvent at the end. `child in children[:]` is forcing to go through all the list element even if it is modifying during iteration in progress. 

Let me know if this makes sense.

Reply:
Neat approach! :-) Thanks again a lot for your support in this, @dakbhavesh!

Reply:
You are welcome @RalphTro!

Reply:
Many thanks for spotting that error and sorry for my ugly implementation of list filtering! https://github.com/RalphTro/epcis-event-hash-generator/commit/7e3aa1cd3e323db1c10e4d2149e7b1a662adf8f5